### ISSUE TITLE
Auth works in powershell core 7.2.1 but not in powershell 5.1

### ISSUE THREAD

**Issue by pierrediancourt** [umn-microsoft-automation/UMN-Google#54]
Hello,

When running this basic code made of the Example.ps1 file provided in this repo  
```powershell
#Requires -RunAsAdministrator

Set-PSRepository -Name PSGallery -InstallationPolicy Trusted
Install-Module UMN-Google
Import-Module UMN-Google

# ref : https://jamesachambers.com/modify-google-sheets-using-powershell/

function Get-WebReqErrorDetails {
	<#
		.SYNOPSIS
			Returns JSON Responsbody data from an Error thrown by Invoke-Webrequest or Invoke-RestMethod
		.DESCRIPTION
			Returns JSON Responsbody data from an Error thrown by Invoke-Webrequest or Invoke-RestMethod
		.PARAMETER err
			Error thrown by Invoke-Webrequest or Invoke-RestMethod
		.NOTES
			Author: Travis Sobeck
	#>
	[CmdletBinding()]
	param
	(
		[Parameter(Mandatory)]
		[System.Management.Automation.ErrorRecord]$err
	)
	
	$reader = New-Object System.IO.StreamReader($err.Exception.Response.GetResponseStream())
	$reader.BaseStream.Position = 0
	$reader.DiscardBufferedData()
	return ($reader.ReadToEnd() | ConvertFrom-Json)
}

# Set security protocol to TLS 1.2 to avoid TLS errors
[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12

# Google API Authozation
# Service token
$scope = "https://www.googleapis.com/auth/spreadsheets https://www.googleapis.com/auth/drive https://www.googleapis.com/auth/drive.file"
$certPath = Resolve-Path "$PSScriptRoot/xxx"
$iss = "xxx" #this is the service account email it will look like  <something>@<projectname>.iam.gserviceaccount.com
$certPswd = "xxx" #Password for certificate
try {
    $accessToken = Get-GOAuthTokenService -scope $scope -certPath $certPath -certPswd $certPswd -iss $iss
} catch {
    (Get-WebReqErrorDetails $_).error_description

    $err = $_.Exception
    $err | Select-Object -Property *
    "Response: "
    $err.Response
}
echo "$accessToken"
```

I get this error   
![image](https://user-images.githubusercontent.com/7483201/151702796-b314e4ac-9b14-4486-ad3a-695b63b8fdb6.png)

But only when i run it in powershell 5.1  
![image](https://user-images.githubusercontent.com/7483201/151702682-633e4caa-c41a-4deb-81e0-bdc0aad87bac.png)
There's no error thrown and the token is displayed when running in PowerShell 7.2.1  
![image](https://user-images.githubusercontent.com/7483201/151702847-560b8691-f0a1-4d8d-b996-ee481dde8bdb.png)
  
  
   
My clock is synced with microsoft time servers as it is the default on any W10 machine.
I can live with running that with the pwsh binary but i thought it should be referenced somewhere :)

Thanks for your work on that project

**Comment by travis-sobeck** [comment]
@pierrediancourt huh.  I have the exact same PS version and run 'Get-GOAuthTokenService' on a regular basis.  Our production automation servers when we created that were 5.x

**Comment by pierrediancourt** [comment]
Alright, fair enough :D 
That's probably an issue caused by something else then.
I'll be glad to update the title of this issue with truer label if we get to the bottom of this mystery !

**Comment by ronnybuchmann** [comment]
the actual issue is the system locale / culture settings
see 3https://stackoverflow.com/a/66022792

**Comment by Breiver7** [comment]
> Alright, fair enough :D That's probably an issue caused by something else then. I'll be glad to update the title of this issue with truer label if we get to the bottom of this mystery !

Hi!
I encountered this issue couple days ago when i've changed date format from MM/dd/yyyy to dd.MM.yyyy in my OS settings.
After that my code similar to above mentioned ceased to work in powershell 5.1 but was still okay when running in PowerShell 7.2.1. 
Ultimately i reverted changes with date format back and code was working in powershell 5.1 again
Don't know if it's related to this case but i hope this is gonna help "to get to the bottom of this mystery" in a way :D

**Comment by kylecweeks** [comment]
Item closed as of module 2.0.2 -> pulled in Ronny's update as the fix. All code tested in Ubuntu LTS 22 and Windows 11 (pwsh 7.4 / powerShell 5.1)
